File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-0.java
class Solution {
    public int removeDuplicates(int[] nums) {
        TreeSet<Integer> set = new TreeSet<Integer>();

        for (int i = 0; i < nums.length; i++) {
            set.add(nums[i]);
        }

        int i = 0;
        for (int num: set) {
            nums[i++] = num;
        }

        return set.size();
    }
}

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-0.kt
class Solution {
    fun findMin(nums: IntArray): Int { 
        var l = 0
        var r = nums.size - 1

        while (l < r) {
            val m = l + (r - l) / 2
            
            if (nums[m] < nums[r]) {
                r = m
            } else {
                l = m + 1
            }
        }

        return nums[l]
    }
}
